src/vs/base/parts/tree/browser/treeView.ts

```diff
@@ -830,8 +830,10 @@ export class TreeView extends HeightMap {
 	}
 
 	public set scrollTop(scrollTop: number) {
+		// {{SQL CARBON EDIT}}
+		let offset = this.domNode.offsetTop ? this.domNode.offsetTop : 0;
```

Comment:
what are we trying to do with `offset` here?  How do we know this doesn't impact the behavior of the tree when used in other contexts?

Reply:
by "top offset" you mean any tree that isn't located at coordinate `(0,0)` with it's container div?

Reply:
@kburtram Yes.

Reply:
@abist could we solve the problem by including the tree in a div and then applying the top offset to the container?  I'd prefer not to change the shared control class for a particular feature if possible, and I know we're able to position the tree not fixed to the top-left corner in some places such as the Remote File Browser dialog.  Please let me know if I'm missing something, but I still don't understand why we need this change?

<img width="191" alt="screen" src="https://user-images.githubusercontent.com/599935/44117549-73ac79be-9fc8-11e8-9345-ab5295939872.png">

Reply:
The scrolling works based on the monaco-scrollable-element, whose height's logic doesn't account for offsetTop. So far, this was the only solution I could find, but I'll try other methods where touching vscode side isn't necessary.

Reply:
@abist why does the domnode have an offset? that node is internal to the class and shouldn't be being modified from external sources.